src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -329,6 +331,35 @@ private Object readResolve() {
             return getDisplayName();
         }
 
+        @Restricted(NoExternalUse.class) // for Jelly
+        public @CheckForNull String getEnclosingLabel() throws Exception {
+            FlowNode executorStepNode = context.get(FlowNode.class);
+            if (executorStepNode != null) {
+                for (StepExecution exec : executorStepNode.getExecution().getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
This feels really iffy to me, but I can't put a finger on why.   But I do wonder if there is a reason why we would want to wait a minute while doing UI rendering?